src/instructlab/model/chat.py

```diff
@@ -0,0 +1,771 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
When you break up a PR into a series of commits, each commit should make sense on its own as a contained step in a progression. This commit adds a ton of code, but doesn't remove any or change any, which doesn't really make sense.

I would like to see a chunk moved in one go. As it stands, I don't think the way it's broken up is any easier to review than the whole thing in a single commit. It still has to be viewed all together to make sense of the moves.

Does that make sense?

Reply:
Another suggestion -- it's often much easier to submit refactorings completely separate from functional changes. It's OK if that's separate commits in a single PR. That would help a lot here. It seems you wanted to do a significant reorganization of the code as part of this. It's best to do that with zero functional changes if possible, and then put the functional changes in their own commits later.

Reply:
I'm trying to dig up some examples of PRs that have a series of commits that can be reviewed independently. Here are a few ...

- https://github.com/instructlab/instructlab-bot/pull/125/commits
- https://github.com/instructlab/instructlab/pull/994/commits
- https://github.com/instructlab/instructlab/pull/951/commits

or some other resources on the topic, linked to specific relevant sections ...

- https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#smaller-is-better-small-commits-small-pull-requests
- https://www.berrange.com/tags/commit-message/
- https://docs.kernel.org/process/submitting-patches.html#separate-your-changes
-